### PR TITLE
feat: generate access token endpoint

### LINKED ISSUE #136: Refresh access token action should reflect status of the ongoing action
### Describe the problem

The Refresh Access Token action doesn't prompt the user when the API call is in process nor any feedback if an error occurred. Find below a screenshot of cases when the token API call fails and there is no feedback; only a console error is shown.
Screenshot 2023-10-19 at 2 55 16 PM

### Proposed solution

When refreshing the token:
- Clear the currently visible token
- Add a message or a loading spinner 
- Show the refreshed token after the Token API call success
- Show an error when Token API fails

### Alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**
- Implemented the tokens API with a cloud function
- Setup basic structure to add more cloud functions in the future
- Add unit tests to the tokens' function.
- Refactor unit test and lint scripts, adding the new python functions.

Closes #136 

**Expected behavior:** 

**Testing tips:**

- Login to DEV environment and copy your refresh token
-  Execute the following curl command:
```
curl --location 'https://api-dev.mobilitydatabase.org/v1/tokens' \
--header 'Content-Type: application/json' \
--data '{
    "refresh_token": "HERE_YOUR_TOKEN"
}'
```
- Verify that you get a 200 HTTP status response with the following structure:
```
{
    "access_token": "ACCESS_TOKEN",
    "expiration_datetime_utc": "2023-11-13T21:46:35Z",
    "token_type": "Bearer"
}
```
- Use the newly generated access token to hit any Feeds API endpoint. Example:
```
curl --location 'https://api-dev.mobilitydatabase.org//v1/metadata' \
--header 'Accept: application/json' \
--header 'Authorization: Bearer HERE_YOUR_ACCESS_TOKEN'
```

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: functions-python/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Python Functions
+
+This folder contains Python functions. These functions should not have interdependencies. However, common code can be shared across multiple functions if they are in folders that don't contain functions.
+
+# Python function folders
+On each function folder, a file should contain all related configurations. Currently, there is no automated process to deploy all functions; this is why, to be able to deploy it the function should be added to the terraform scripts. In the future, we should axe to automate the deployment.
```
Comment (jcpitre) [inline functions-python/README.md:6]: Wrong word:
we should `axe` to automate the deployment

**Review by qcdyx — APPROVED** [review]
This is fantastic!

File: scripts/function-python-run.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#
+
+# Executes the python function passed as argument supported by the functions-framework.
+# The script receives the name of the function as parameter.
+# The function must be located in the folder `functions-python/<function_name>`.
+# The function must be defined in the file `functions-python/<function_name>/main.py`.
+# The script look for a `.env.local` file in the function's folder to load environment variables.
+
+# Usage:
+#   run-python-function.sh --function_name <function name>
```
Comment (jcpitre) [inline scripts/function-python-run.sh:26]: function-python-run.sh

File: scripts/function-python-build.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+# Build a python function compressing the source code and it's dependencies exluding the libs in requirements.txt.
```
Comment (jcpitre) [inline scripts/function-python-build.sh:19]: typo: `its`

**Review by jcpitre — APPROVED** [review]

**Pull request merged**